Updated SDK-->Application Cli - Closes 1540 (#1621)
LiskHQ/lisk-docs: modules/ROOT/pages/application-cli.adoc

[source,bash]
----
Lisk-SDK Application

VERSION
  dpos-mainchain/0.1.0 darwin-arm64 node-v16.16.0

USAGE
  $ ./bin/run --help

TOPICS
  account        Commands relating to dpos-mainchain accounts.
  block          Commands relating to dpos-mainchain blocks.
  blockchain     Commands relating to dpos-mainchain blockchain data.
  blskey         Return bls public & private keys corresponding to the given passphrase.
  config         Commands relating to dpos-mainchain node configuration.
  endpoint       Invokes the provided endpoint.
  generator      Commands relating to dpos-mainchain generation and generator-info data.
  genesis-block  Creates a genesis block file.
  keys           Commands relating to account, generator, and bls keys.
  node           Commands relating to dpos-mainchain node.
  passphrase     Commands relating to dpos-mainchain passphrases.
  transaction    Commands relating to dpos-mainchain transactions.

COMMANDS
  autocomplete  Display autocomplete installation instructions

>   console       Lisk interactive REPL session to run commands.

Author: Agreed, text updated.

[source,bash]
----
Lisk-SDK Application

VERSION
  dpos-mainchain/0.1.0 darwin-arm64 node-v16.16.0

USAGE
  $ ./bin/run [COMMAND] --help

TOPICS
  account        Commands relating to dpos-mainchain accounts.
  block          Commands relating to dpos-mainchain blocks.
  blockchain     Commands relating to dpos-mainchain blockchain data.
  blskey         Returns the bls public & private keys corresponding to the given passphrase.
  config         Commands relating to dpos-mainchain node configuration.

>   forger-info    Commands relating to dpos-mainchain forger-info data.

Author: Agreed, text updated

>   forging        Commands relating to dpos-mainchain forging.

Author: Agreed, text updated

  $ ./bin/run start

OPTIONS
  -c, --config=config                                  File path to a custom config. Environment variable "LISK_CONFIG_FILE" can also be used.

>   -d, --data-path=data-path                            Directory path to specify where node data is stored. Environment variable "LISK_DATA_PATH" can also be used.

Reviewer: it doesn't not default to `~/.lisk/dpos-mainchain/data/`. 
data path is `os.homedir()` `.lisk`, `application name`

                                                       connection. The environment variable 
                                                       "LISK_MONITOR_PLUGIN_WHITELIST" can also be used.

  --overwrite-config                                   Overwrite network configs if they exist already.

  --seed-peers=seed-peers                              Seed peers to initially connect to in format of comma 
                                                       separated "ip:port". IP can be DNS name or IPV4 format. 
                                                       The environment variable "LISK_SEED_PEERS" can also be used.

EXAMPLES
  ./bin/run start
  ./bin/run start --network devnet --data-path /path/to/data-dir --log debug
  ./bin/run start --network devnet --api-ws
  ./bin/run start --network devnet --api-ws --api-ws-port 8888
  ./bin/run start --network devnet --port 9000
  ./bin/run start --network devnet --port 9002 --seed-peers 127.0.0.1:9001,127.0.0.1:9000
  ./bin/run start --network testnet --overwrite-config
  ./bin/run start --network testnet --config ~/my_custom_config.json
----
.Example
[%collapsible]
====
[source,bash]
----

> $ ./bin/run start --api-ipc --api-http

Author: I see that you have a strong opinion about it, may I know why? as the existing example is correct and it also shows the use of two parameters to the reader. It tells the reader that you can start your server with `--api-ipc` and `--api-http` together as well. I also see similar usage in SDK threads and repositories so why change that?

> $ ./bin/run start --api-ipc --api-http

Author: Agreed, text updated.

2022-09-12T14:42:36.469Z INFO xyz.local application 24009 [method=random_getHashOnionSeeds] Registered endpoint
2022-09-12T14:42:36.469Z INFO xyz.local engine 24010 Engine started
2022-09-12T14:42:36.469Z INFO xyz.local application 24009 [method=random_hasHashOnion] Registered endpoint
2022-09-12T14:42:36.469Z INFO xyz.local application 24009 [method=random_getHashOnionUsage] Registered endpoint
2022-09-12T14:42:36.469Z INFO xyz.local application 24009 [method=dpos_getVoter] Registered endpoint
2022-09-12T14:42:36.469Z INFO xyz.local application 24009 [method=dpos_getDelegate] Registered endpoint
2022-09-12T14:42:36.469Z INFO xyz.local application 24009 [method=dpos_getAllDelegates] Registered endpoint
2022-09-12T14:42:36.470Z INFO xyz.local application 24009 [method=dpos_getConstants] Registered endpoint
2022-09-12T14:42:36.470Z INFO xyz.local application 24009 [method=interoperability_getInteroperabilityStore] Registered endpoint
2022-09-12T14:42:37.565Z INFO xyz.local engine 24010 [id=8e9dae6b2c91e0114473347105ee9c9c9f989324126c09633a0ef05b6bc57b96 height=115 generatorAddress=lskxa4895zkxjspdvu3e5eujash7okvnkkpr8xsr5] Generated new block
----
====

=== console

> Provides the Lisk interactive {url_read_eval_print_loop}[REPL] session to run commands.

Author: Agreed, text updated.

  $ ./bin/run config:COMMAND

COMMANDS

>   ./bin/run config:create  Creates network configuration file.

Author: Agree, text updated

OPTIONS
  -i, --community-identifier=community-identifier  [default: sdk] Community Identifier
  -l, --label=label                                [default: beta-sdk-app] App Label

>   -o, --output=output                              [default: /Users/iamtalha/Documents/GitHub/lisk-sdk/examples/dpos-mainchain] Directory where the config file is saved

Author: Agreed, text updated

>   -o, --output=output                              [default: /Users/iamtalha/Documents/GitHub/lisk-sdk/examples/dpos-mainchain] Directory where the config file is saved

Author: Agreed, text updated

OPTIONS
  -d, --data-path=data-path  Directory path to specify where node data is stored. Environment variable "LISK_DATA_PATH" can also be used.

>   -o, --output=output        The output directory. Default will set to current working directory.

Author: Ummmm, I have copied/pasted the contents for all application CLI commands since I wasn't sure about the change. Hence, the original statements have been changed. Should we reword these descriptions? I didn't do it because I wanted to keep the content synchronous with the application CLI and the documentation.

>   -o, --output=output        The output directory. Default will set to current working directory.

Reviewer 2: I think most of them look fine. You could maybe just check them again, and in addition, I will also have another look 
 on the next review round 👍

>   -o, --output=output        The output directory. Default will set to current working directory.

Author: Cool, Will take another look :)

----
USAGE
  $ ./bin/run genesis-block:COMMAND

COMMANDS
  ./bin/run genesis-block:create  Creates genesis block file.
----

==== genesis-block:create
Creates a genesis block file and the corresponding delegate\'s configuration data and accounts information.

[source,bash]
----
USAGE
  $ ./bin/run genesis-block:create

OPTIONS
  -a, --accounts=accounts                                                                    [default: 10] Number of non-validator accounts to generate.
  -c, --config=config                                                                        File path to a custom config. Environment variable "LISK_CONFIG_FILE" can also be used.
  -f, --assets-file=assets-file                                                              Path to the file which contains the genesis block asset in JSON format.
  -n, --network=network                                                                      [default: default] Default network config to use. Environment variable "LISK_NETWORK" can also be used.
  -o, --output=output                                                                        [default: config] Output folder path of the generated genesis block.

>   -t, --token-distribution=token-distribution                                                [default: 100000000000] Amount of tokens distributed to each account.

Author: cc: @shuse2 @ishantiw

OPTIONS
  -d, --data-path=data-path  Directory path to specify where node data is stored. Environment variable "LISK_DATA_PATH" can also be used.
  --pretty                   Prints JSON in a pretty format rather than condensed.

EXAMPLES
  ./bin/run node:info
  ./bin/run node:info --data-path ./lisk
----

==== node:metadata
Gets a node\'s metadata from a running blockchain application.

[source,bash]
----
USAGE

>   $ ./bin/run node:metadata

Author: Faced the same issue at my end, awaiting for advise regarding this issue.

  $ ./bin/run transaction:COMMAND

COMMANDS
  ./bin/run transaction:create  Create transaction which can be broadcasted to the network. Note: fee and amount should be in Beddows!!
  ./bin/run transaction:get     Get transaction from local node by ID.
  ./bin/run transaction:send    Send transaction to the local node.
  ./bin/run transaction:sign    Sign encoded transaction.
----

==== transaction:create

> Creates a transaction which can be broadcasted to the network.

Reviewer 2: ```suggestion
Creates a transaction that can be broadcasted to the network.
```

  COMMAND  Registered transaction command.
  FEE      Transaction fee in Beddows.

OPTIONS
  -a, --params=params                            Creates transaction with specific params information
  -d, --data-path=data-path                      Directory path to specify where node data is stored. Environment variable "LISK_DATA_PATH" can also be used.

  -f, --file=file                                The file to upload.
                                                        Example:
                                                                --file=./myfile.json

  -j, --json                                     Print the transaction in JSON format.

  -k, --key-derivation-path=key-derivation-path  [default: m/25519'/134'/0'/0'] Key derivation path to use to derive keypair from passphrase

  -p, --passphrase=passphrase                    Specifies a source for your secret passphrase. Command will prompt you for input if this option is not set.
                                                        Examples:
                                                        - --passphrase='my secret passphrase' (should only be used where security is not important)

>   -s, --sender-public-key=sender-public-key      Creates the transaction with provided sender publickey, when passphrase is not provided

Reviewer 2: ```suggestion
  -s, --sender-public-key=sender-public-key      Creates the transaction with the provided sender publickey, when the passphrase is not provided
```

>   --network-identifier=network-identifier        Network identifier defined for the network or main | test for the Lisk Network.

Author: How does this look?

```suggestion
  --network-identifier=network-identifier        A network identifier for Lisk. The accepted value is either "main" or "test".
```

>   --network-identifier=network-identifier        Network identifier defined for the network or main | test for the Lisk Network.

Reviewer 2: Sounds good 👍

  --no-signature                                 Creates the transaction without a signature. Your passphrase will therefore not be required.

  --nonce=nonce                                  Nonce of the transaction.

  --offline                                      Specify whether to connect to a local node or not.

  --pretty                                       Prints JSON in a pretty format rather than condensed.

EXAMPLES

>   ./bin/run transaction:create token transfer 100000000 --params='{"amount":100000000,"recipientAddress":"ab0041a7d3f7b2c290b5b834d46bdc7b7eb85815","data":"send token"}'

Author: Agreed, text updated.

OPTIONS
  -d, --data-path=data-path  Directory path to specify where node data is stored. Environment variable "LISK_DATA_PATH" can also be used.
  --pretty                   Prints JSON in a pretty format rather than condensed.

EXAMPLE
  ./bin/run transaction:get eab06c6a22e88bca7150e0347a7d976acd070cb9284423e6eabecd657acc1263
----

==== transaction:send
Sends a transaction to the local node.

[source,bash]
----
USAGE
  $ ./bin/run transaction:send TRANSACTION

ARGUMENTS
  TRANSACTION  A transaction to be sent to the node encoded as hex string

OPTIONS
  -d, --data-path=data-path  Directory path to specify where node data is stored. Environment variable "LISK_DATA_PATH" can also be used.
  --pretty                   Prints JSON in a pretty format rather than condensed.

EXAMPLE
  ./bin/run transaction:send 

>   080810011880cab5ee012220fd061b9146691f3c56504be051175d5b76d1b1d0179c5c4370e18534c58821222a2408641214ab0041a7d3f7b2c290b5b834d46bdc7b7eb858151a0a73656e6420746f6b656e324028edd3601cdc35a41bb23415a0d9f3c3e9cf188d9971adf18742c

Author: Not sure, maybe something happened during copy/paste.

      "height": 802,
      "maxHeightGenerated": 646,
      "maxHeightPrevoted": 733
    },
    {
      "address": "lskf5sf93qyn28wfzqvr74eca3tywuuzq6xf32p7f",
      "height": 827,
      "maxHeightGenerated": 809,
      "maxHeightPrevoted": 755
    },
    {
      "address": "lskfowbrr5mdkenm2fcg2hhu76q3vhs74k692vv28",
      "height": 725,
      "maxHeightGenerated": 625,
      "maxHeightPrevoted": 628
    },
    {
      "address": "lskf7a93qr84d9a6ga543wernvxbsrpvtp299c5mj",
      "height": 792,
      "maxHeightGenerated": 706,
      "maxHeightPrevoted": 723
    },
    {
      "address": "lskfjd3ymhyzedgneudo2bujnm25u7stu4qpa3jnd",
      "height": 843,
      "maxHeightGenerated": 769,
      "maxHeightPrevoted": 763
    },
    {
      "address": "lskffxs3orv2au2juwa69hqtrmpcg9vq78cqbdjr4",
      "height": 826,
      "maxHeightGenerated": 818,
      "maxHeightPrevoted": 755
    },
    {
      "address": "lskgn7m77b769frqvgq7uko74wcrroqtcjv7nhv95",
      "height": 848,
      "maxHeightGenerated": 761,
      "maxHeightPrevoted": 764
    }
  ]
}
----
====

==== generator:import
Imports generator\'s data from a file.
[source,bash]
----
USAGE
  $ ./bin/run generator:import

OPTIONS

>   -d, --data-path=data-path  Directory path to specify where node data is stored. The environment variable "LISK_DATA_PATH" 

Reviewer 2: ```suggestion
  -d, --data-path=data-path  Directory path to specify where the node data is stored. The environment variable "LISK_DATA_PATH" 
```

                             can also be used.

  -f, --file-path=file-path  (required) Path of the file to import from

  --pretty                   Prints JSON in pretty format rather than condensed.

EXAMPLES
  ./bin/run generator:import --file-path ./my/path/genInfo.json
  ./bin/run generator:import --file-path ./my/path/genInfo.json --data-path ./data 
----

.Example 
[%collapsible]
====
[source,bash]
----
$./bin/run generator:import --file-path ~/mypath/geninfo.json
$
----
====

==== generator:status
Gets block generation information for the locally running node.
[source,bash]
----
USAGE
  $ ./bin/run generator:status

OPTIONS

>   -d, --data-path=data-path  Directory path to specify where node data is stored. The environment variable "LISK_DATA_PATH" 

Reviewer 2: ```suggestion
  -d, --data-path=data-path  Directory path to specify where the node data is stored. The environment variable "LISK_DATA_PATH" 
```
